packages/dx-react-scheduler-material-ui/src/templates/appointment-form/recurrence/weekly-recurrence-selector.jsx

```diff
@@ -0,0 +1,181 @@
+import React from 'react';
+import Button from '@material-ui/core/Button';
+import ButtonGroup from '@material-ui/core/ButtonGroup';
+import * as PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import classNames from 'classnames';
+import {
+  getRecurrenceOptions, DAYS_OF_WEEK, SUNDAY_DATE, WEEK_DAY_OPTIONS, MONDAY_DATE,
+  TUESDAY_DATE, THURSDAY_DATE, FRIDAY_DATE, SATURDAY_DATE, WEDNESDAY_DATE,
+  handleWeekDaysChange, changeRecurrenceOptions,
+} from '@devexpress/dx-scheduler-core';
+import { setColor } from '../../utils';
+
+const styles = ({ palette, spacing }) => ({
+  selectedButton: {
+    backgroundColor: setColor(400, palette.primary),
+    '&:hover': {
+      backgroundColor: setColor(500, palette.primary),
+    },
+    border: `1px solid ${setColor(400, palette.primary)}!important`,
+    borderLeft: `1px solid ${setColor(50, palette.primary)}!important`,
+    '&:first-child': {
+      borderLeft: `1px solid ${setColor(400, palette.primary)}!important`,
+    },
+    color: setColor(50, palette.primary),
+  },
+  button: {
+    minWidth: spacing(3),
+  },
+  buttonGroup: {
+    marginBottom: spacing(2),
+  },
+});
+
+const isCurrentWeekDay = (recurrenceOptions, weekDay) => recurrenceOptions.byweekday
+  && recurrenceOptions.byweekday.indexOf(weekDay) > -1;
+
+const WeeklyRecurrenceSelectorBase = React.memo(({
+  formatDate,
+  rRule,
+  readOnly,
+  classes,
+  className,
+  onFieldChange,
+  ...restProps
+}) => {
+  const recurrenceOptions = getRecurrenceOptions(rRule);
+  return (
+    <ButtonGroup
+      variant="outlined"
+      size="small"
+      disabled={readOnly}
+      className={classNames(classes.buttonGroup, className)}
+      fullWidth
+      {...restProps}
+    >
+      <Button
```

Comment:
It can be something like:
```
array.map((el) => {
    return (
        <Button
        className={classNames({
          [classes.button]: true,
          [classes.selectedButton]: isCurrentWeekDay(recurrenceOptions, el[0]),
        })}
        onClick={() => onFieldChange({
          rRule: changeRecurrenceOptions(handleWeekDaysChange(
            recurrenceOptions,
            el[0],
            onFieldChange,
          )),
        })}
      >
        { formatDate(el[1], WEEK_DAY_OPTIONS) }
      </Button>
    )
}),
```
where `array = [[DAYS_OF_WEEK.SUNDAY, SUNDAY_DATE], [DAYS_OF_WEEK.MONDAY, DAYS_OF_WEEK.MONDAY], ...]`